app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -695,56 +698,80 @@ class HomeFragment : Fragment(), CoroutineScope, AccountObserver {
         emitAccountChanges()
     }
 
-    private fun scrollToCollections(addedTabsSize: Int) {
+    private fun scrollAndAnimateCollection(addedTabsSize: Int, changedCollection: TabCollection? = null) {
         launch {
+            val recyclerView = sessionControlComponent.view
             delay(ANIM_SCROLL_DELAY)
             val tabsSize = requireComponents.core.sessionManager.sessions.filter {
                 (activity as HomeActivity).browsingModeManager.isPrivate == it.private
             }.size
-            sessionControlComponent.view.smoothScrollToPosition(tabsSize + NON_TAB_ITEM_NUM)
-            delay(ANIM_SNACKBAR_DELAY)
-            showSavedSnackbar(addedTabsSize)
+            var indexOfCollection = tabsSize + NON_TAB_ITEM_NUM
+            changedCollection?.let { changedCollection ->
+                requireComponents.core.tabCollectionStorage.cachedTabCollections.filterIndexed { index, tabCollection ->
+                    if (tabCollection.id == changedCollection.id) {
+                        indexOfCollection = tabsSize + NON_TAB_ITEM_NUM + index
+                        true
+                    } else {
+                        false
+                    }
+                }
+            }
+            val lastVisiblePosition =
+                (recyclerView.layoutManager as? LinearLayoutManager)?.findLastCompletelyVisibleItemPosition() ?: 0
+            if (lastVisiblePosition < indexOfCollection) {
```

Comment:
How does this work if the lastVisiblePosition was off screen *below* indexOfCollection? Does it handle scrolling up and down?